doc/developer-guides/interrupt-hld.rst

```diff
@@ -451,6 +451,13 @@ a guest's vMSI. The handler then injects the corresponding virtual
 interrupt into the guest, based on vPIC/vIOAPIC/vLAPIC APIs described
 earlier.
 
+Interrupt Storm Mitigation: When device model launch a User OS, ACRN will
+remap interrupt for this user OS's pass-through devices. So when an interrupt
+occurs for such device, the CPU core that is assigned to that User OS gets
+trapped into hypervisor. The benefit of such mechanism is that should an
+interrupt storm happen in a particular UOS, it will have little effect on
+the performance of the Service OS.
+
```

Comment:
I think the rational is that the processing of software_IRQ  for the UOS will take global lock, which will block the processing of software_IRQ in SOS side.  Because previously we use global lock...  Is it the case?

Reply:
yes, correct. my another patch is to remove the global lock, replaced by a per VM lock. So the explanation for our design is matched.

Reply:
Are you saying we should hold merging this documentation PR until some code change happens?

Reply:
The patch will be merged soon. So, no worry, we can keep revising this patch.